synedrion/src/paillier/encryption.rs

```diff
@@ -160,40 +171,75 @@ impl<P: PaillierParams> Ciphertext<P> {
     }
 
     /// Encrypts the plaintext with the provided randomizer.
-    pub fn new_with_randomizer(pk: &PublicKeyPaillier<P>, plaintext: &P::Uint, randomizer: &RandomizerWire<P>) -> Self {
+    pub fn new_with_randomizer(
+        pk: &PublicKeyPaillier<P>,
+        plaintext: &Secret<P::Uint>,
+        randomizer: &Randomizer<P>,
+    ) -> Self {
         Self::new_with_randomizer_inner(pk, plaintext, randomizer, Choice::from(0))
     }
 
+    pub fn new_with_randomizer_bounded(
```

Comment:
Maybe `new_with_bounded_randomizer`? Reads a bit better. Same for the signed version.

Reply:
`new_with_bounded_randomizer` implies that the randomized is `Bounded`, but it's not the case - the distinguishing feature is that the secret is `Bounded`.

Reply:
Of course, 🤦 .

But in that case, how about "new_bounded_with_randomizer"?

Reply:
That would imply that the result is Bounded. I am not a big fan of the name myself, but a grammatically correct variant "new_with_randomizer_and_bounded_secret" seems to be a little long?

Reply:
Yep, you're right, that's too long. Let's keep the current name then.